swap/src/asb/event_loop.rs

```diff
@@ -899,15 +914,42 @@ mod quote {
     pub fn unreserved_monero_balance(
         unlocked_balance: Amount,
         reserved_amounts: impl Iterator<Item = Amount>,
+        developer_tip: Option<Decimal>,
     ) -> Amount {
+        use rust_decimal::prelude::ToPrimitive;
+
+        // If a developer tip is configured, we need to account for the fact that
+        // to lock X XMR for a swap, we actually need X * (1 + tip_percentage) XMR
+        // because the tip is sent as an additional output in the same transaction
+
+        // To find how much we can actually use for swaps, we need to solve:
+        //     swap_amount * multiplier <= available_after_reserved
+        // <=> swap_amount <= available_after_reserved / multiplier
+
+        // Calculate the effective multiplier: 1 + tip_percentage
```

Comment:
Clearer variable names would be: total_piconero_balance, total_reserved_piconero, available_piconero, tip_adjusted_available_piconero